proposals/NNNN-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -0,0 +1,176 @@
+  ***Note** for SDL *PROTOCOL* version higher than or equal 4.4 in in StartServiceACK response Proxy can get *MAJOR*.*MINOR*.*PATCH* versions*
+  ***Note** since that moment (StartServiceACK response) and until RegisterAppInterface successful response Proxy should use PROTOCOL version to filter out all unknown enum values*
+
+  So the workflow should be implemented as follows:
+  1. The Mobile Application (Android/IOS) creates a proxy object (proxy) to communicate with SDL core.
+  2. The Mobile Application passes to the Proxy entity the configurations.
+  3. Among other properties the configuration object has the following parameters:
+      * minimal SDL version
+      * application HMI type(s) (`AppHMIType` array)
+  #### Changes related to sending request from Mobile application to SDL
+  After the connection is established and Mobile Application is registered with SDL core, Mobile Application works as usual but requires the request parameters validation. An issue will occur when the Mobile Application calls a method and passes an enum values array as a parameter that belongs to the higher SDL version. In this case, the Mobile Proxy should filter out unsupported params before sending the request and provide warning in logs as `unsupported_values`.
+  The validation sequence:
+  1. Mobile Application sends request to SDL core through the proxy.
+  2. The proxy checks the request parameters:
+     * If there are no enum values unknown to SDL, the proxy forwards the request to SDL core without changes.
+     
+     * If some of the enum values are unknown to SDL core the proxy should:
+          * filter out all RPC’s unsupported enum values including nested RPC’s if those are present, before making a request to make sure it is compliant with the current spec
+          * log on warning level with «unsupported_values» (enum’s/struct’s names)
+    
+  **Note**: For `RegisterAppInterface` to filter we should use version of PROTOCOL, but not the MOBILE_API according to system’s flow. **iOS** *should be extended with similar mechanism in order to encapsulate filtering logic.*
+
+  **For example** : 
+  If the UI app requested unsupported `AppHMIType` and version which are not compatible then the proxy must filter unsupported HMI types. 
+
+      - Before filtering :
+          "AppHMIType" : ["DEFAULT", "MEDIA", "UKNOWN"]
+      - After filtering :
+          AppHMIType" : ["DEFAULT", "MEDIA"]
+  ***Note**: In case no HMI types are left after filtering, Mobile library should throw Exception, otherwise RegisterAppInterface RPC will be sent with remaining HMI types.*
+  ***Note**: Now system flow is the following: if `RegisterAppInterface` fails with an error then Mobile library should pass the error status to the Mobile Library adding failure reason*
+  ***Note**: The protocol version control logic partly implemented in Android and not implemented in iOS though it has its flaws and should be updated following the logic above.*
```

Comment:
```suggestion
***Note**: In case no HMI types are left after filtering, Mobile Library should throw Exception, otherwise `RegisterAppInterface` RPC will be sent with remaining HMI types.*
  
***Note**: Now system flow is the following: if `RegisterAppInterface` fails with an error then Mobile library should pass the error status to the Mobile Library adding failure reason*
  
***Note**: The protocol version control logic partly implemented in Android and not implemented in iOS though it has its flaws and should be updated following the logic above.*
```

I'm not sure I follow the last note here.  Should it be updated to "The protocol version control logic is partly implemented in Android and not implemented in iOS. However, what is implemented has its flaws and should be updated following the logic above."?

Reply:
Please address my question here, and revise line 203 accordingly.

Reply:
@IGapchuk can you please answer my question here?

Reply:
@theresalech Yes, you are right. Your rephrased variant explains our thoughts about the described approach. We will rewrite this sentence. Thank you!